docs/architecture/speculative-workflow-task.md

```diff
@@ -6,157 +6,168 @@ There are three types of Workflow Task:
   2. Transient
   3. Speculative
 
-Every Workflow Task ships history to the worker. Last events must be `WorkflowTaskScheduled`
-and `WorkflowTaskStarted`. There might be some events in between if they come after Workflow Task
-was scheduled but not started (e.g., Workflow worker was down and didn't poll for Workflow Task).
-
-**Normal Workflow Task** is created by server every time when server needs Workflow to make progress.
-If Workflow Task fails (worker responds with call to `RespondWorkflowTaskFailed` or an error occurred while
-processing `RespondWorkflowTaskCompleted`) or times out (worker is disconnected),
-server writes corresponding Workflow Task failed event in the history and increase attempt count
-in the mutable state. For the next attempt, Workflow Task events (`WorkflowTaskScheduled`
-and `WorkflowTaskStarted`) are not written into the history but attached to the response of
-`RecordWorkflowTaskStarted` API. These are transient Workflow Task events. Worker is not aware of any "transience"
-of these events. If Workflow Task keeps failing, attempt counter is getting increased in mutable state,
-but no new fail events are written into the history and new transient Workflow Task events are just recreated.
-Workflow Task, which has transient Workflow Task events, is called **transient Workflow Task**.
-When Workflow Task finally completes, `WorkflowTaskScheduled` and `WorkflowTaskStarted` events
-are getting written to history followed by `WorklfowTaskCompleted` event.
+Every Workflow Task ships history events to the worker. It must always contain the two events
+`WorkflowTaskScheduled` and `WorkflowTaskStarted` (must be last!). There might be some events in-between
+them if they came in after the Workflow Task was scheduled but not yet started (e.g. when the Workflow
+worker was down and didn't poll for Workflow Task).
+
+A **normal Workflow Task** is created by the server when it needs a Workflow to make progress. If 
+the Workflow Task fails (i.e. worker responds with a call to `RespondWorkflowTaskFailed` or an error
+occurred while processing `RespondWorkflowTaskCompleted`) or times out (e.g. worker is disconnected),
+the server writes a corresponding Workflow Task failed event to the history and increases the
+attempt count in the mutable state.
+
+For the next attempt, **transient Workflow Task events** are used: the Workflow Task events
+`WorkflowTaskScheduled` and `WorkflowTaskStarted` are *not* written to the history, but attached to
+the response from the `RecordWorkflowTaskStarted` API. The worker does not know they are transient,
+though. If the Workflow Task keeps failing, the attempt counter is increased in the mutable state,
+and transient Workflow Task events are created again - but no new failure event is written into the
+history again. When the Workflow Task finally completes, the `WorkflowTaskScheduled` and
```

Comment:
Now you removed `transient WFT` completely :-). But we do use this name in code and comments. I would add an original sentence here:
```
Workflow Task, which has transient Workflow Task events, is called ***transient Workflow Task***.
```